core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -242,6 +245,53 @@ private ManifestFile copyManifest(ManifestFile manifest) {
         current.formatVersion(), toCopy, current.specsById(), newManifestPath, snapshotId(), appendedManifestsSummary);
   }
 
+  /**
+   * Validates that no files matching given partitions have been added to the table since a starting snapshot.
+   *
+   * @param base table metadata to validate
+   * @param startingSnapshotId id of the snapshot current at the start of the operation
+   * @param partitionSet a set of partitions to check against, or none if check is to be against all files
+   */
+  protected void validateAddedDataFiles(TableMetadata base, Long startingSnapshotId,
```

Comment:
Rather than using an `Optional<PartitionSet>` couldn't this use `validateAddedDataFiles` with filter `Expressions.alwaysTrue()`? That would make this a bit more straightforward. and avoid the `isPresent` check inside a loop (which probably isn't a good idea anyway).

Reply:
Done.  only thing is it makes it back to a bad error message "Found conflicting files that can contain records matching true", if that's ok